### PR TITLE
GAGS-ifies ties, and removes inherent ties from some outfits.

### PR DESCRIPTION
**Pull request by Wallemations**
## About The Pull Request
Removes the inherent ties from most civilian outfits with a single-color tie baked into them.

But why?

Well, that's because the default Tie has been made into a GAGS sprite!
tally hall

Now you can live out your dreams of fashionable neckwear using any color you desire!

All ties now start untied, and you have to take 4 seconds to manually tie them. Clumsy people have a bit of a problem doing this, but the command staff have become so used to it they can do it in half the time! Mirrors help to speed this up even further!

## Why It's Good For The Game
The fact that a lot of suit sprites have the ties baked into them makes the existing tie item a bit redundant.
Furthermore, using GAGS makes it so a player not only has more customization options, but now there's a nice untied icon state for that "I just woke up" look!

## Changelog

:cl: Wallem
imageadd: Ties have now been GAGS-ified
add: Ties now must be manually tied, allowing for both a classic and a "I just woke up" look.
imagedel: Removed inherent ties from several outfits.
/:cl:

### CODE REVIEW

**Review by JohnFulpWillard — CHANGES_REQUESTED** [review]
Since you're removing ties from the suits, please put them in the appropriate vendors/closets to replace them (Lawyer/RD).

**Comment by Wallemations** [comment]
> Since you're removing ties from the suits, please put them in the appropriate vendors/closets to replace them (Lawyer/RD).

At the moment this only touches some civilian outfits, and even that was hard to find candidates for because a lot of these suits also have the jacket baked in too.

Since RD is being resprited as we speak I didn't touch it, I was asked not to touch the ties security has baked in, and the detective's outfits don't have ties baked-in.

File: code/modules/clothing/neck/_neck.dm [hunk 1]
```diff
@@ -18,36 +18,116 @@
...
+			H.update_clothing(ITEM_SLOT_NECK)
+	register_context()
+
+/obj/item/clothing/neck/tie/greyscale/AltClick(mob/user)
+	. = ..()
+	to_chat(user, span_notice("You concentrate as you begin [is_tied ? "untying" : "tying"] [src]..."))
+	// Heads of staff are experts at tying their ties.
+	var/tie_timer_actual = tie_timer
+	if(user.mind.assigned_role.departments_bitflags & DEPARTMENT_BITFLAG_COMMAND)
+		tie_timer_actual = tie_timer_actual / 2
```
Comment (LemonInTheDark) [inline code/modules/clothing/neck/_neck.dm:57]: Please document this important balance change

Comment (Wallemations) [reply code/modules/clothing/neck/_neck.dm:57]: Right, forgot that bit, give me a bit to add it to the main thing

Comment (Wallemations) [reply code/modules/clothing/neck/_neck.dm:57]: done

**Comment by Wallemations** [comment]
and now that I did this the map errors'll arrive, time to fix those real quick too!

**Comment by JohnFulpWillard** [comment]
@Wallemations Ties in the lawyer vendor and RD's locker is what I meant, since their suits dont have the ties anymore

**Comment by Wallemations** [comment]
> @Wallemations Ties in the lawyer vendor and RD's locker is what I meant, since their suits dont have the ties anymore

Yeye, I'm going through and doing that now.

File: code/modules/clothing/neck/_neck.dm [hunk 2]
```diff
@@ -18,36 +18,119 @@
...
+/obj/item/clothing/neck/tie/greyscale/AltClick(mob/user)
+	. = ..()
+	to_chat(user, span_notice("You concentrate as you begin [is_tied ? "untying" : "tying"] [src]..."))
+	var/tie_timer_actual = tie_timer
+	// Mirrors give you a boost to your tying speed. I realize this stacks and I think that's hilarious.
+	for(var/obj/structure/mirror/reflection in view(2, user))
+		tie_timer_actual = tie_timer_actual / 1.25
+	// Heads of staff are experts at tying their ties.
+	if(user.mind.assigned_role.departments_bitflags & DEPARTMENT_BITFLAG_COMMAND)
+		tie_timer_actual = tie_timer_actual / 2
```
Comment (MrMelbert) [inline code/modules/clothing/neck/_neck.dm:60]:
```suggestion
		tie_timer_actual /= 2
```
el operator

File: code/modules/clothing/neck/_neck.dm [hunk 3]
```diff
@@ -18,36 +18,119 @@
...
+			H.update_clothing(ITEM_SLOT_NECK)
+	register_context()
+
+/obj/item/clothing/neck/tie/greyscale/AltClick(mob/user)
+	. = ..()
+	to_chat(user, span_notice("You concentrate as you begin [is_tied ? "untying" : "tying"] [src]..."))
+	var/tie_timer_actual = tie_timer
+	// Mirrors give you a boost to your tying speed. I realize this stacks and I think that's hilarious.
+	for(var/obj/structure/mirror/reflection in view(2, user))
+		tie_timer_actual = tie_timer_actual / 1.25
```
Comment (MrMelbert) [inline code/modules/clothing/neck/_neck.dm:57]:
```suggestion
		tie_timer_actual /= 1.25
```

File: code/modules/clothing/neck/_neck.dm [hunk 4]
```diff
@@ -18,36 +18,119 @@
...
+	// Heads of staff are experts at tying their ties.
+	if(user.mind.assigned_role.departments_bitflags & DEPARTMENT_BITFLAG_COMMAND)
```
Comment (MrMelbert) [inline code/modules/clothing/neck/_neck.dm:59]:
```suggestion
	if(user.mind?.assigned_role.departments_bitflags & DEPARTMENT_BITFLAG_COMMAND)
```
It's very unlikely but not impossible mindless mobs will be tying ties

File: code/modules/clothing/neck/_neck.dm [hunk 5]
```diff
@@ -18,36 +18,119 @@
...
+	if((HAS_TRAIT(user, TRAIT_CLUMSY) || HAS_TRAIT(user, TRAIT_DUMB)) && prob(50))
+		to_chat(user, span_notice("You just can't seem to get a proper grip on [src]!"))
+		return
+	// Success!
+	is_tied = !is_tied
+	visible_message("[user] adjusts [user.p_their()] tie [HAS_TRAIT(user, TRAIT_BALD) ? "." : "and slicks back [user.p_their()] hair."]", span_notice("You successfully [is_tied ? "tied" : "untied"] [src]!"))
+	update_appearance(UPDATE_ICON)
+	if(ishuman(usr))
+		var/mob/living/carbon/human/H = usr
+		H.update_clothing(ITEM_SLOT_NECK)
```
Comment (MrMelbert) [inline code/modules/clothing/neck/_neck.dm:75]: Three things:
one, `user` not `usr`
two, one letter var names
three, `update_clothing` is mob level proc anyways, why cast to human?

File: code/modules/clothing/neck/_neck.dm [hunk 6]
```diff
@@ -18,36 +18,119 @@
...
+	if(!do_after(user, tie_timer_actual))
+		to_chat(user, span_notice("Your fingers fumble away from [src] as your concentration breaks."))
+		return
+	// Clumsy & Dumb people have trouble tying their ties.
+	if((HAS_TRAIT(user, TRAIT_CLUMSY) || HAS_TRAIT(user, TRAIT_DUMB)) && prob(50))
+		to_chat(user, span_notice("You just can't seem to get a proper grip on [src]!"))
+		return
+	// Success!
+	is_tied = !is_tied
+	visible_message("[user] adjusts [user.p_their()] tie [HAS_TRAIT(user, TRAIT_BALD) ? "." : "and slicks back [user.p_their()] hair."]", span_notice("You successfully [is_tied ? "tied" : "untied"] [src]!"))
```
Comment (MrMelbert) [inline code/modules/clothing/neck/_neck.dm:71]: `user.visible_messaage`?

Comment (MrMelbert) [inline code/modules/clothing/neck/_neck.dm:71]: Also, I'm not 100% sure all bald people have `TRAIT_BALD`.

File: code/modules/clothing/neck/_neck.dm [hunk 7]
```diff
@@ -18,36 +18,119 @@
...
+	/// How long it takes to tie the tie
+	var/tie_timer = 4 SECONDS
+
+/obj/item/clothing/neck/tie/greyscale/Initialize(mapload)
+	. = ..()
+	update_appearance(UPDATE_ICON)
+	if(usr)
+		if(ishuman(usr))
+			var/mob/living/carbon/human/H = usr
+			H.update_clothing(ITEM_SLOT_NECK)
```
Comment (MrMelbert) [inline code/modules/clothing/neck/_neck.dm:48]: what the - is `usr` set for `Initialize`??

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/modules/clothing/neck/_neck.dm (hunk 2 above)
Comment (Wallemations) [reply code/modules/clothing/neck/_neck.dm:60]: For some reason I didn't think the operator worked, oops

File: code/modules/clothing/neck/_neck.dm (hunk 5 above)
Comment (Wallemations) [reply code/modules/clothing/neck/_neck.dm:75]: I'm gonna be honest I just copied this bit from the jumpsuit adjusting code, since the tie worn overlay wasn't updating properly

File: code/modules/clothing/neck/_neck.dm (hunk 7 above)
Comment (Wallemations) [reply code/modules/clothing/neck/_neck.dm:48]: How else am I supposed to use `update_clothing(ITEM_SLOT_NECK)` if the tie started around someone's neck?

File: code/modules/clothing/neck/_neck.dm (hunk 6 above)
Comment (Wallemations) [reply code/modules/clothing/neck/_neck.dm:71]: that's fair, but I didn't want to do an entire `var/mob/living/carbon/human/bald_guy = user` to make a check for the user's current hairstyle.

Comment (Wallemations) [reply code/modules/clothing/neck/_neck.dm:71]: oops!

File: code/modules/clothing/neck/_neck.dm [hunk 8]
```diff
@@ -18,36 +18,125 @@
...
+	var/is_tied = FALSE
+	/// How long it takes to tie the tie
+	var/tie_timer = 4 SECONDS
+
+/obj/item/clothing/neck/tie/greyscale/Initialize(mapload)
+	. = ..()
+	update_appearance(UPDATE_ICON)
+	if(ismob(loc))
+		var/mob/wearer = loc
+		wearer.update_clothing(ITEM_SLOT_NECK)
```
Comment (LemonInTheDark) [inline code/modules/clothing/neck/_neck.dm:47]: why is this required exactly?

Comment (Wallemations) [reply code/modules/clothing/neck/_neck.dm:47]: If it starts spawned in a player's inventory it needs to update its icon

**Comment by Jolly-66** [comment]
This is just my inner mapper complaining if you can include an update paths script.
It would be nice and make somethings easier later down the line as a retrospective ordeal.

If you don't know how to write them, check the tools/UpdatePaths folder, there's a plethora of examples.
The most basic being;
`old_path : new_path`

File: code/modules/clothing/neck/_neck.dm (hunk 8 above)
Comment (LemonInTheDark) [reply code/modules/clothing/neck/_neck.dm:47]: When would it spawn in the player's inventory? shouldn't it be equipped to the relevant slot later?
This pattern worries me

**Comment by Wallemations** [comment]
> This is just my inner mapper complaining if you can include an update paths script.

Ok

**Comment by Wallemations** [comment]
assuming I've got most of the major problems fixed, I'm going to un-draft this, since it's more or less everything I had planned for this

File: code/modules/clothing/neck/_neck.dm (hunk 8 above)
Comment (Wallemations) [reply code/modules/clothing/neck/_neck.dm:47]: outfits like the psychologist's main one come to mind, as well as some others.

Comment (Wallemations) [reply code/modules/clothing/neck/_neck.dm:47]: if this is a problematic thing, I can just set the tied versions of the tie to have the correct icon_state without needing their appearance updated
The reason that update is there is because of the vending machines. On the vending machine, the tie visually displays as tied because its `icon_state` is set to tied, but when it's dispensed, it updates to its proper untied state. If they start off as tied, they actually wouldn't even need this updated icon state, and I realize now as I type this that it's a completely redundant thing.

**Review by Imaginos16 — APPROVED** [review]
image

File: code/modules/clothing/neck/_neck.dm [hunk 9]
```diff
@@ -18,36 +18,126 @@
...
+	if(!do_after(user, tie_timer_actual))
+		to_chat(user, span_notice("Your fingers fumble away from [src] as your concentration breaks."))
+		return
+	// Clumsy & Dumb people have trouble tying their ties.
+	if((HAS_TRAIT(user, TRAIT_CLUMSY) || HAS_TRAIT(user, TRAIT_DUMB)) && prob(50))
+		to_chat(user, span_notice("You just can't seem to get a proper grip on [src]!"))
+		return
+	// Success!
+	is_tied = !is_tied
+	user.visible_message("[user] adjusts [user.p_their()] tie[HAS_TRAIT(user, TRAIT_BALD) ? "." : " and runs a hand across [user.p_their()] head."]", span_notice("You successfully [is_tied ? "tied" : "untied"] [src]!"))
```
Comment (MrMelbert) [inline code/modules/clothing/neck/_neck.dm:67]: The first parameter of this visible_message has no span.
Also, some indentation can help here. 
```suggestion
	user.visible_message(
		span_notice("[user] adjusts [user.p_their()] tie[HAS_TRAIT(user, TRAIT_BALD) ? "." : " and runs a hand across [user.p_their()] head."]"),
		span_notice("You successfully [is_tied ? "tied" : "untied"] [src]!"),
	)
```

File: code/modules/clothing/neck/_neck.dm [hunk 10]
```diff
@@ -18,36 +18,126 @@
...
+		strip_delay = 4 SECONDS
+		equip_delay_other = 4 SECONDS
+		equip_delay_self = 2 SECONDS
+	else // Extremely quick strip delay, it's practically a ribbon draped around your neck
+		icon_state = "tie_greyscale_untied"
+		strip_delay = 1 SECONDS
+		equip_delay_other = 1 SECONDS
+		equip_delay_self = 0
+
+/obj/item/clothing/neck/tie/greyscale/add_context(atom/source, list/context, obj/item/held_item, mob/user)
```
Comment (MrMelbert) [inline code/modules/clothing/neck/_neck.dm:85]: The context is cool, maybe also put the hotkey into `examine()` for those who have it disabled?

File: code/modules/clothing/neck/_neck.dm [hunk 11]
```diff
@@ -18,36 +18,126 @@
...
+	desc = "The necktie is adorned with a garish pattern. It's disturbingly vivid. Somehow you feel as if it would be wrong to ever take it off. It's your friend now. You will betray it if you change it for some boring scarf."
+
 /obj/item/clothing/neck/tie/detective
 	name = "loose tie"
 	desc = "A loosely tied necktie, a perfect accessory for the over-worked detective."
 	icon_state = "detective"
 
+/obj/item/clothing/neck/tie/detective/examine(mob/user)
+	. = ..()
+	. += "Looking closely, you can see that it's actually a cleverly disguised clip-on."
```
Comment (MrMelbert) [inline code/modules/clothing/neck/_neck.dm:139]: (This also has no span, but if you want it to use the default examine span then that's also ok)

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/modules/clothing/neck/_neck.dm (hunk 11 above)
Comment (Wallemations) [reply code/modules/clothing/neck/_neck.dm:139]: this one was an intentional one, it's just meant to be an addition on a newline

File: code/modules/clothing/neck/_neck.dm (hunk 9 above)
Comment (Wallemations) [reply code/modules/clothing/neck/_neck.dm]: done

File: code/modules/clothing/neck/_neck.dm (hunk 10 above)
Comment (Wallemations) [reply code/modules/clothing/neck/_neck.dm:95]: done

**Comment by Wallemations** [comment]
Hold on don't merge this I just figured out how to make non-greyscale subtypes

**Comment by Wallemations** [comment]
ok all set

**Comment by Wallemations** [comment]
that didn't work hold on I gotta manually change these DMIs
